packages/migrate/index.ts

```diff
@@ -74,129 +95,144 @@ module.exports = function migrate(...args) {
  * user decides to abort the migration
  */
 
-function runMigration(currentConfigPath, outputConfigPath) {
-	const recastOptions = {
-		quote: "single"
+function runMigration(currentConfigPath: string, outputConfigPath: string): Promise<void> | void {
+	const recastOptions: object = {
+		quote: "single",
 	};
-	const tasks = new Listr([
+
+	const tasks: Listr = new Listr([
 		{
-			title: "Reading webpack config",
-			task: ctx =>
-				new PLazy((resolve, reject) => {
+			task: (ctx: INode): string | void | Listr | Promise<{}> =>
+				new pLazy((
+						resolve: (value?: object) => void,
+						reject: (reason?: string | object, err?: object) => void,
+					) => {
 					fs.readFile(currentConfigPath, "utf8", (err, content) => {
 						if (err) {
 							reject(err);
 						}
 						try {
-							const jscodeshift = require("jscodeshift");
+							const jscodeshift: IJSCodeshift = require("jscodeshift");
 							ctx.source = content;
 							ctx.ast = jscodeshift(content);
 							resolve();
 						} catch (err) {
 							reject("Error generating AST", err);
 						}
 					});
-				})
+				}),
+			title: "Reading webpack config",
 		},
 		{
-			title: "Migrating config to newest version",
-			task: ctx => {
-				const transformations = require("./migrate").transformations;
+			task: (ctx: INode): string | void | Listr | Promise<any> => {
 				return new Listr(
-					Object.keys(transformations).map(key => {
-						const transform = transformations[key];
+					Object.keys(transformations).map((key: string): {
+						task: (_?: void) => string;
+						title: string;
+					} => {
```

Comment:
missing ret for map here

Reply:
Oh, I think you missed it. It's already present. The object with properties `task` and `title` is the return type.

```js
.map((key: string): {
    task: (_?: void) => string;
    title: string;
}